ledger-api/grpc-definitions/com/daml/ledger/api/v1/event_query_service.proto

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "com/daml/ledger/api/v1/value.proto";
+
+option java_outer_classname = "EventQueryServiceOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// Allow clients to query events
+service EventQueryService {
+
+  // Get the create and the consuming exercise event for the contract with the provided ID
+  rpc GetEventsByContractId (GetEventsByContractIdRequest) returns (GetEventsByContractIdResponse);
+
+  // Get all create and consuming exercise events for the contracts with the provided contract key
+  rpc GetEventsByContractKey (GetEventsByContractKeyRequest) returns (GetEventsByContractKeyResponse);
+
+}
+
+message GetEventsByContractIdRequest {
+  // The contract id being queried.
+  // Required
+  string contract_id = 1;
+
+  // The parties whose events the client expects to see.
+  // The events associated with the contract id will only be returned if the requesting parties includes
+  // at least one party that is a stakeholder of the event. For a definition of stakeholders see
+  // https://docs.daml.com/concepts/ledger-model/ledger-privacy.html#contract-observers-and-stakeholders
+  // Required
+  repeated string requesting_parties = 2;
+
+}
+
+message GetEventsByContractIdResponse {
+
+  // Will be populated if a created event is found for the request.
+  // Optional
+  CreatedEvent create_event = 1;
```

Comment:
It has just occurred to me that if in the future we will elect to send more data with an event, such as its offset, its projection, it won't be an easy extension. We can trivially protect ourselves from it by enclosing an event in a message already. wdyt?

Reply:
My thoughts are:
1. This interface is the minimal needed to fill the missing gap (between contract id/key -> event)
2. We can go event -> transaction to get all tx info if requrired
3. If in the future we needed some extra info we could have `create_meta`/`archive_meta` (at top level).

Reply:
:+1: